src/ol_infrastructure/applications/concourse/__main__.py

```diff
@@ -708,13 +709,62 @@ def build_worker_user_data(
         user_data=consul_datacenter.apply(build_worker_user_data_partial),
     )
 
+    # We will create a 'fake' lb, targetgroup and lblistener to make use of
+    # aws's ability to do healthchecks and automatically swap out stalled
+    # concourse workers.
+    # We will never actually interact with workers via these resources.
+    worker_lb = lb.LoadBalancer(
+        f"concourse-worker-{worker_class_name}-load-balancer",
+        internal=True,
+        ip_address_type="dualstack",
+        load_balancer_type="application",
+        name=f"concourse-worker-alb-{worker_class_name[:3]}-{stack_info.env_suffix[:2]}",
+        security_groups=[concourse_worker_security_group.id],
+        subnets=target_vpc["subnet_ids"],
+        tags=aws_config.merged_tags({}),
+    )
+
+    worker_target_group = lb.TargetGroup(
+        f"concourse-worker-{worker_class_name}-target-group",
+        vpc_id=ops_vpc_id,
+        port=CONCOURSE_WORKER_HEALTHCHECK_PORT,
+        protocol="HTTP",
+        health_check=lb.TargetGroupHealthCheckArgs(
+            healthy_threshold=2,
+            interval=60,
+            matcher="200",
+            path="/",
+            port=CONCOURSE_WORKER_HEALTHCHECK_PORT,
+            protocol="HTTP",
+            timeout=20,
+            unhealthy_threshold=5,
+        ),
+        name=f"concourse-worker-tg-{worker_class_name[:3]}-{stack_info.env_suffix[:2]}"[
+            :TARGET_GROUP_NAME_MAX_LENGTH
+        ],
+        tags=aws_config.merged_tags(worker_def["aws_tags"]),
+    )
+
+    worker_lb_alb_listener = lb.Listener(
```

Comment:
Do we need a listener configured since we won't actually be routing any traffic to the workers?

Reply:
Maybe if we just add the workers as a target group to the existing Concourse ALB and just don't use a listener since we don't actually care about (or necessarily want?) traffic making it to the workers.

Reply:
Roger that. Maybe we can scope the rules for the listener to only accept traffic on the 10. address range so that we don't get arbitrary traffic hitting those endpoints?

Reply:
So, you raise interesting questions and I did science. 

1. A normal elb address from the web instances:
![Screen Shot 2022-02-14 at 12 47 31 PM](https://user-images.githubusercontent.com/41908054/153918903-a7bde195-a136-4352-861e-bcb3e8bcaf72.png)

2. One of the worker elb addresses:
![Screen Shot 2022-02-14 at 12 45 53 PM](https://user-images.githubusercontent.com/41908054/153918951-130801f1-6ba3-4895-ae2f-11acb760e219.png)

So the web ELB instance returns _public_ addresses of concourse-web nodes. The 'internal' (line 718 in this file), returns the private IP addresses from the CI ops vpc. It is interesting that it can be resolved from anywhere (assuming you could guess the whole name), but it resolves to an address that is not route-able outside of the VPC.